text/python/0000-poetry.md

```diff
@@ -0,0 +1,177 @@
+
+#### Alternative:
+
+It is possible that the `poetry` buildpack could download the install script
+and `poetry` release artifact of a specific version to separate layers and run
+the install script.  Fortunately, the `get-poetry.py` script
+[supports](https://github.com/python-poetry/poetry/pull/2162) a `--file` flag
+that takes the path of an archive from which to install the `poetry` cli. Thus,
+the buildpack could run `python get-poetry.py --file <release-asset.tgz>` with
+`POETRY_HOME=<path/to/target/layer>` set.
+
+### Dependency Installation
+
+By default, poetry creates virtualenvs to install dependencies into. This
+[config
+option](https://python-poetry.org/docs/configuration/#virtualenvsin-project),
+when set to `true`, causes poetry to create a virtualenv inside the project
+directory.
+
+Once a virtualenv is created, a user might run `poetry shell` which
+"activates" the virtualenv, followed by `poetry install` and `poetry
+run`. This activation is less useful in the context of a running container
+since the environment has already been optimally configured by the buildpacks.
+The poetry-install buildpack will therefore forego the `poetry shell` step and
+instead:
+
+1. Configure poetry to create the virtualenv within the project dir
+1. Run `poetry install`
+
+The resulting virtualenv will be stored in a separate layer and symlinked into
+the working directory.
+
+### Launching an app
+
+Unlike the existing order groups in the Python buildpack, the `poetry` group
+will rely less on Procfile in favor of a more idiomatic `poetry-run` buildpack.
+Poetry has its own `run` command and it seems worthwhile to sacrifice
+consistency with the other Python buildpacks for a more idiomatic user
+experience. The alternative would be to specify `poetry run` as part of a
+process type in a Procfile.
+
+## Prior Art
+
+- The Miniconda buildpack downloads an installation script which installs
+  `conda` [RFC000X]. To support this change, `packit` was modified to handle
+  downloading text files/scripts as dependency.
+
+
+## Unresolved Questions and Bikeshedding
+
+- Where will packages be installed?
+    - It does not seem that installing to a custom directory is officially
+      supported yet. (Poetry Issues:
+      [#2003](https://github.com/python-poetry/poetry/issues/2003) &
+      [#1937](https://github.com/python-poetry/poetry/issues/1937))
+
+- What steps are needed to enable caching with poetry?
+
+  - In the Node.js buildpack, npm-install runs the dependency installation
+    process from the working directory and moves the resulting files into a
+    separate layer. This new location is subsequently symlinked to the relevant
+    path within the working directory. This approach could be reused in the
+    poetry-install buildpack.
+
+  - How does the native poetry cache interact with buildpack caching?
+
+- Should there be a poetry-run-script buildpack?
+
+## Future Topics
+
+- Should `poetry-install` provide `site-packages`?
```

Comment:
The primary reason we would want to do this is to provide a standardized interface across all of the package managers. If there is a use-case we can imagine where we'd want this type of interoperability, its worth considering.